scanner/src/funTest/assets/file-counter-expected-output-for-analyzer-result.yml

```diff
@@ -215,8 +215,7 @@ scanner:
     - "**/META-INF/DEPENDENCIES"
   results:
     scan_results:
-    - id: "Gradle:org.ossreviewtoolkit.gradle.example:lib:1.0.0"
-      results:
+      Gradle:org.ossreviewtoolkit.gradle.example:lib:1.0.0:
```

Comment:
Just some remarks:
- From just looking at the YAML as a human, this is now slightly harder to read as the is is not quoted anymore, and ends with another `:`.
- From an "API" design perspective, this goes exactly the opposite way as recently explained by @pombredanne [here](https://github.com/nexB/vulnerablecode/issues/284#issuecomment-783346856) ("there should be no value used as a key").

Despite saying that, I *do* like the resulting code simplifications.

Reply:
Shouldn't we not worry about API design at all since the map will be removed in the next refactoring step anyway again?
(e.g. it's a temporary step isn't it?)

Reply:
> it's a temporary step isn't it?

Yes, with scan-by-repository the key will be the provenance, so expect this will become a list again, because provenance is not a good map key.

> as a human, this is now slightly harder to read

Didn't we stop optimizing for human readability, at least since https://github.com/oss-review-toolkit/ort/pull/3502?

> this goes exactly the opposite way as recently explained by @pombredanne

I think I would agree when it's about a web API, but I'm not so sure if this applies to our result file format as well.

Reply:
> > as a human, this is now slightly harder to read
> 
> Didn't we stop optimizing for human readability, at least since #3502?

We don't *optimize* for human readability, but we still should have an eye on keeping a reasonable balance. If we can very cheaply make something much more human readable, we should do that. I'm not saying that's the case here, though.